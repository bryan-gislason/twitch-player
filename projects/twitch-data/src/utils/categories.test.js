import { extractCategories } from ".";

describe("extractCategories", () => {
  const { data } = require("../../fixtures/topCategories.json");
  const similarData = {
    ...data,
    categories: {
      ...data.categories,
      edges: data.categories.edges.slice(),
    },
  };
  const dissimilarData = {
    ...data,
    categories: {
      ...data.categories,
      edges: data.categories.edges.slice(1),
    },
  };

  it("extracts the category nodes from a response", () => {
    expect(extractCategories(data)).toMatchSnapshot();
  });

  it("returns the same result for the exact same argument", () => {
    expect(extractCategories(data)).toBe(extractCategories(data));
  });

  it("returns the same result for similar argument", () => {
    expect(extractCategories(data)).toBe(extractCategories(similarData));
  });

  it("returns a different result for dissimilar argument", () => {
    expect(extractCategories(data)).not.toBe(extractCategories(dissimilarData));
  });
});
