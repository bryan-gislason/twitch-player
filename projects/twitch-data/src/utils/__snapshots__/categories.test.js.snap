// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractCategories extracts the category nodes from a response 1`] = `
Array [
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/Fortnite-376x500.jpg",
    "followersCount": 35800760,
    "id": "33214",
    "key": "33214",
    "name": "Fortnite",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "isLanguageTag": false,
        "key": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "localizedName": "Shooter",
        "tagName": "Shooter",
      },
    ],
    "viewersCount": 241208,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/League%20of%20Legends-376x500.jpg",
    "followersCount": 16964605,
    "id": "21779",
    "key": "21779",
    "name": "League of Legends",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "12510423-d1f6-4992-8aea-1441a43d1df4",
        "isLanguageTag": false,
        "key": "12510423-d1f6-4992-8aea-1441a43d1df4",
        "localizedName": "MOBA",
        "tagName": "MOBA",
      },
    ],
    "viewersCount": 78952,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/Just%20Chatting-376x500.jpg",
    "followersCount": 957935,
    "id": "509658",
    "key": "509658",
    "name": "Just Chatting",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "2610cff9-10ae-4cb3-8500-778e6722fbb5",
        "isLanguageTag": false,
        "key": "2610cff9-10ae-4cb3-8500-778e6722fbb5",
        "localizedName": "IRL",
        "tagName": "IRL",
      },
    ],
    "viewersCount": 56211,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/Atlas-376x500.jpg",
    "followersCount": 77425,
    "id": "5945",
    "key": "5945",
    "name": "Atlas",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "857fce92-df0c-4c8c-8c60-28e0c5f7d8ae",
        "isLanguageTag": false,
        "key": "857fce92-df0c-4c8c-8c60-28e0c5f7d8ae",
        "localizedName": "Strategy",
        "tagName": "Strategy",
      },
      Object {
        "__typename": "Tag",
        "id": "22e434b6-ca88-46e8-91ef-c18ee1cb8a67",
        "isLanguageTag": false,
        "key": "22e434b6-ca88-46e8-91ef-c18ee1cb8a67",
        "localizedName": "Simulation",
        "tagName": "Simulation",
      },
      Object {
        "__typename": "Tag",
        "id": "5495bcba-0977-4ba5-aa39-8928fd4cab8f",
        "isLanguageTag": false,
        "key": "5495bcba-0977-4ba5-aa39-8928fd4cab8f",
        "localizedName": "MMORPG",
        "tagName": "MMORPG",
      },
    ],
    "viewersCount": 56142,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/Dota%202-376x500.jpg",
    "followersCount": 7237025,
    "id": "29595",
    "key": "29595",
    "name": "Dota 2",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "12510423-d1f6-4992-8aea-1441a43d1df4",
        "isLanguageTag": false,
        "key": "12510423-d1f6-4992-8aea-1441a43d1df4",
        "localizedName": "MOBA",
        "tagName": "MOBA",
      },
    ],
    "viewersCount": 29925,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/Sea%20of%20Thieves-376x500.jpg",
    "followersCount": 945779,
    "id": "490377",
    "key": "490377",
    "name": "Sea of Thieves",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "4d1eaa36-f750-4862-b7e9-d0a13970d535",
        "isLanguageTag": false,
        "key": "4d1eaa36-f750-4862-b7e9-d0a13970d535",
        "localizedName": "Action",
        "tagName": "Action",
      },
    ],
    "viewersCount": 27843,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/Counter-Strike:%20Global%20Offensive-376x500.jpg",
    "followersCount": 14531233,
    "id": "32399",
    "key": "32399",
    "name": "Counter-Strike: Global Offensive",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "a69f7ffb-ddda-4c05-8d7d-f0b24975a2c3",
        "isLanguageTag": false,
        "key": "a69f7ffb-ddda-4c05-8d7d-f0b24975a2c3",
        "localizedName": "FPS",
        "tagName": "FPS",
      },
      Object {
        "__typename": "Tag",
        "id": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "isLanguageTag": false,
        "key": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "localizedName": "Shooter",
        "tagName": "Shooter",
      },
    ],
    "viewersCount": 26289,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/Escape%20From%20Tarkov-376x500.jpg",
    "followersCount": 985626,
    "id": "491931",
    "key": "491931",
    "name": "Escape From Tarkov",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "22e434b6-ca88-46e8-91ef-c18ee1cb8a67",
        "isLanguageTag": false,
        "key": "22e434b6-ca88-46e8-91ef-c18ee1cb8a67",
        "localizedName": "Simulation",
        "tagName": "Simulation",
      },
      Object {
        "__typename": "Tag",
        "id": "a69f7ffb-ddda-4c05-8d7d-f0b24975a2c3",
        "isLanguageTag": false,
        "key": "a69f7ffb-ddda-4c05-8d7d-f0b24975a2c3",
        "localizedName": "FPS",
        "tagName": "FPS",
      },
      Object {
        "__typename": "Tag",
        "id": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "isLanguageTag": false,
        "key": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "localizedName": "Shooter",
        "tagName": "Shooter",
      },
    ],
    "viewersCount": 23755,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/Overwatch-376x500.jpg",
    "followersCount": 13837211,
    "id": "488552",
    "key": "488552",
    "name": "Overwatch",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "a69f7ffb-ddda-4c05-8d7d-f0b24975a2c3",
        "isLanguageTag": false,
        "key": "a69f7ffb-ddda-4c05-8d7d-f0b24975a2c3",
        "localizedName": "FPS",
        "tagName": "FPS",
      },
      Object {
        "__typename": "Tag",
        "id": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "isLanguageTag": false,
        "key": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "localizedName": "Shooter",
        "tagName": "Shooter",
      },
    ],
    "viewersCount": 21036,
  },
  Object {
    "__typename": "Game",
    "boxArtURL": "categories/PLAYERUNKNOWN%27S%20BATTLEGROUNDS-376x500.jpg",
    "followersCount": 24944299,
    "id": "493057",
    "key": "493057",
    "name": "PLAYERUNKNOWN'S BATTLEGROUNDS",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "isLanguageTag": false,
        "key": "523fe736-fa95-44c7-b22f-13008ca2172c",
        "localizedName": "Shooter",
        "tagName": "Shooter",
      },
      Object {
        "__typename": "Tag",
        "id": "a69f7ffb-ddda-4c05-8d7d-f0b24975a2c3",
        "isLanguageTag": false,
        "key": "a69f7ffb-ddda-4c05-8d7d-f0b24975a2c3",
        "localizedName": "FPS",
        "tagName": "FPS",
      },
    ],
    "viewersCount": 20884,
  },
]
`;
